tests/unit/ui/QueryEditor.test.tsx

```diff
@@ -138,72 +134,89 @@ describe('QueryEditor RAW', () => {
-    render(<QueryEditor datasource={mockDatasource} query={query} onRunQuery={onRunQuery} onChange={onChange} />);
-
-    expect(autocompleterRequest).toHaveBeenCalledWith(
-      'ajax_vs_autocomplete.py',
-      expect.objectContaining({ ident: 'sites' })
+    const r = render(
+      <QueryEditor datasource={mockDatasource} query={query} onRunQuery={onRunQuery} onChange={onChange} />
     );
-    expect(autocompleterRequest).toHaveBeenCalledWith(
-      'ajax_vs_autocomplete.py',
-      expect.objectContaining({ ident: 'monitored_hostname' })
+
+    expect(contextAutocomplete).toHaveBeenCalledWith('sites', expect.anything(), expect.anything(), expect.anything());
+    expect(contextAutocomplete).toHaveBeenCalledWith(
+      'monitored_hostname',
+      expect.anything(),
+      expect.anything(),
+      expect.anything()
     );
-    expect(autocompleterRequest).toHaveBeenCalledWith(
-      'ajax_vs_autocomplete.py',
-      expect.objectContaining({ ident: 'monitored_service_description' })
+    expect(contextAutocomplete).toHaveBeenCalledWith(
+      'monitored_service_description',
+      expect.anything(),
+      expect.anything(),
+      expect.anything()
     );
-    expect(autocompleterRequest).toHaveBeenCalledWith(
-      'ajax_vs_autocomplete.py',
-      expect.objectContaining({ ident: 'available_graphs' })
+    expect(contextAutocomplete).toHaveBeenCalledWith(
+      'available_graphs',
+      expect.anything(),
+      expect.anything(),
+      expect.anything()
     );
-    autocompleterRequest.mockClear();
+    contextAutocomplete.mockClear();
 
-    const siteInput = screen.getByLabelText('Site');
+    const siteInput = r.getByLabelText('Site');
     await act(async () => {
       await selectEvent.select(siteInput, 'Site One', { container: document.body });
     });
-    expect(autocompleterRequest).toHaveBeenCalledWith(
-      'ajax_vs_autocomplete.py',
-      expect.objectContaining({ ident: 'monitored_hostname' })
+    expect(contextAutocomplete).toHaveBeenCalledWith(
+      'monitored_hostname',
+      expect.anything(),
+      expect.anything(),
+      expect.anything()
     );
-    expect(autocompleterRequest).toHaveBeenCalledWith(
-      'ajax_vs_autocomplete.py',
-      expect.objectContaining({ ident: 'monitored_service_description' })
+    expect(contextAutocomplete).toHaveBeenCalledWith(
+      'monitored_service_description',
+      expect.anything(),
+      expect.anything(),
+      expect.anything()
     );
-    expect(autocompleterRequest).toHaveBeenCalledWith(
-      'ajax_vs_autocomplete.py',
-      expect.objectContaining({ ident: 'available_graphs' })
+    expect(contextAutocomplete).toHaveBeenCalledWith(
+      'available_graphs',
+      expect.anything(),
+      expect.anything(),
+      expect.anything()
     );
-    autocompleterRequest.mockClear();
+    contextAutocomplete.mockClear();
 
     const hostInput = screen.getByLabelText('Hostname');
```

Comment:
Should screen be replaced with r here as well? As you did in line 161/162

Reply:
i changed it back to screen, in favor of a smaller patch.